Python/ceval.c

```diff
@@ -1249,14 +1247,13 @@ eval_frame_handle_pending(PyThreadState *tstate)
 #ifdef Py_STATS
 #define INSTRUCTION_START(op) \
     do { \
-        frame->f_lasti = INSTR_OFFSET(); \
-        next_instr++; \
+        frame->next_instr = ++next_instr; \
         OPCODE_EXE_INC(op); \
         _py_stats.opcode_stats[lastopcode].pair_count[op]++; \
         lastopcode = op; \
     } while (0)
 #else
-#define INSTRUCTION_START(op) frame->f_lasti = INSTR_OFFSET(); next_instr++
+#define INSTRUCTION_START(op) (frame->next_instr = ++next_instr)
```

Comment:
Here's something to consider. 

Rather that storing `next_instr`, we could store `last_instr` (strictly, `next_instr_less_one`) as it would break the dependency of the store on the addition.
`frame->next_instr_less_one = next_instr; next_instr++`
The compiler can then merge the `next_instr++` with any subsequent `JUMPBY(INLINE_CACHE_ENTRIES_...)`

It would also simplify the changes as you wouldn't need to change all the offsets by one.

Thoughts?

Reply:
https://github.com/brandtbucher/cpython/compare/lasti...brandtbucher:lasti-alt

Reply:
The performance numbers show no difference between that branch and this one. The other branch does result in a net reduction of "adjust by 1" moves, so I think I slightly prefer it.

Thoughts?

Reply:
I prefer the other version. It should be quicker, and feels less intrusive. I'm not surprised that we there is no measurable difference in performance, I would expect it to be very slight.